controllers/cloud.redhat.com/helpers/namespaces.go

```diff
@@ -19,66 +19,70 @@ import (
 	// WORKAROUND: Can't set annotations and ownerref on project request during create
 	// Performing annotation and ownerref change in one transaction
-	namespace, err := GetNamespace(ctx, cl, namespace.Name)
+	ns, err := GetNamespace(ctx, cl, nsName)
 	if err != nil {
-		return namespace.Name, fmt.Errorf("could not retrieve newly created namespace [%s]: %w", namespace.Name, err)
+		return ns, fmt.Errorf("could not retrieve newly created namespace [%s]: %w", nsName, err)
 	}
 
-	utils.UpdateAnnotations(&namespace, CreateInitialAnnotations())
-	utils.UpdateLabels(&namespace, CreateInitialLabels(pool.Name))
-	namespace.SetOwnerReferences([]metav1.OwnerReference{pool.MakeOwnerReference()})
+	utils.UpdateAnnotations(&ns, CreateInitialAnnotations())
+	utils.UpdateLabels(&ns, CreateInitialLabels(pool.Name))
+	ns.SetOwnerReferences([]metav1.OwnerReference{pool.MakeOwnerReference()})
 
-	if err := cl.Update(ctx, &namespace); err != nil {
-		return namespace.Name, fmt.Errorf("could not update Project [%s]: %w", namespace.Name, err)
+	if err := cl.Update(ctx, &ns); err != nil {
+		return ns, fmt.Errorf("could not update Project [%s]: %w", nsName, err)
 	}
 
 	// Create ClowdEnvironment
-	if err := CreateClowdEnv(ctx, cl, pool.Spec.ClowdEnvironment, namespace.Name); err != nil {
-		return "", fmt.Errorf("error creating ClowdEnvironment for namespace [%s]: %w", namespace.Name, err)
+	if err := CreateClowdEnv(ctx, cl, pool.Spec.ClowdEnvironment, nsName); err != nil {
+		return ns, fmt.Errorf("error creating ClowdEnvironment for namespace [%s]: %w", nsName, err)
 	}
 
 	// Create LimitRange
 	limitRange := pool.Spec.LimitRange
-	limitRange.SetNamespace(namespace.Name)
+	limitRange.SetNamespace(nsName)
 
 	if err := cl.Create(ctx, &limitRange); err != nil {
-		return "", fmt.Errorf("error creating LimitRange for namespace [%s]: %w", namespace.Name, err)
+		return ns, fmt.Errorf("error creating LimitRange for namespace [%s]: %w", nsName, err)
 	}
 
 	// Create ResourceQuotas
 	resourceQuotas := pool.Spec.ResourceQuotas
 	for _, quota := range resourceQuotas.Items {
 		innerQuota := quota
-		innerQuota.SetNamespace(namespace.Name)
+		innerQuota.SetNamespace(nsName)
 		if err := cl.Create(ctx, &innerQuota); err != nil {
-			return "", fmt.Errorf("error creating ResourceQuota for namespace [%s]: %w", namespace.Name, err)
+			return ns, fmt.Errorf("error creating ResourceQuota for namespace [%s]: %w", nsName, err)
 		}
 	}
 
 	// Copy secrets
-	if err := CopySecrets(ctx, cl, namespace.Name); err != nil {
-		return "", fmt.Errorf("error copying secrets from ephemeral-base namespace to namespace [%s]: %w", namespace.Name, err)
+	if err := CopySecrets(ctx, cl, nsName); err != nil {
+		return ns, fmt.Errorf("error copying secrets from ephemeral-base namespace to namespace [%s]: %w", nsName, err)
 	}
 
-	return namespace.Name, nil
+	return ns, nil
```

Comment:
Another great refactor. This new method makes much more sense than having one method do all this stuff. I do think this could be pulled out into more stuf, but overall its much better now than it was.